### PR TITLE
Improve documentation for kubeadm phases

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR is part of the effort for improving kubeadm reference doc (CLI, website, manpages), and more specifically improves documentation for:
- kubeadm alpha phase certs
- kubeadm alpha phase kubeconfig
- kubeadm alpha phase controlplane
- kubeadm alpha phase etcd
- kubeadm alpha phase uploadconfig
- kubeadm alpha phase addons
- kubeadm alpha phase selfhosting

**Which issue(s) this PR fixes**:
part of the effort for [#265](https://github.com/kubernetes/kubeadm/issues/265)

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/uploadconfig.go [hunk 1]
```diff
@@ -22,18 +22,36 @@ import (
...
 	"k8s.io/kubernetes/cmd/kubeadm/app/phases/uploadconfig"
 	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
 	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	uploadConfigLongDesc = normalizer.LongDesc(`
+		Uploads the kubeadm init configuration of your cluster to a ConfigMap called kubeadm-config in the kube-system namespace.
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/uploadconfig.go]: nit: can this be parameterized from `constants.go`?

File: cmd/kubeadm/app/cmd/phases/selfhosting.go [hunk 2]
```diff
@@ -34,15 +34,33 @@ import (
...
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	selfhostingLongDesc = normalizer.LongDesc(`
+		Converts static Pod files for control plane components into self-hosted DaemonSets configured via the Kubernetes API.
+
+		Self-hosting in 1.8 has some important limitations. See documentation.
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/selfhosting.go]: Self-hosting in v1.8 is on an alpha level with some limitations, but on a beta level in v1.9. See the documentation

File: cmd/kubeadm/app/cmd/phases/selfhosting.go [hunk 3]
```diff
@@ -34,15 +34,33 @@ import (
 	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	selfhostingLongDesc = normalizer.LongDesc(`
+		Converts static Pod files for control plane components into self-hosted DaemonSets configured via the Kubernetes API.
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/selfhosting.go]: nit: Static Pod

File: cmd/kubeadm/app/cmd/phases/kubeconfig.go [hunk 4]
```diff
@@ -28,14 +28,58 @@ import (
...
+		# Generates all kubeconfig files, functionally equivalent to what generated
+		# by kubeadm init.
+		kubeadm alpha phase kubeconfig all
+
+		# Generates all kubeconfig files using options read from a configuration file.
+		kubeadm alpha phase kubeconfig all --config masterconfiguration.yaml
+		`)
+
+	adminKubeconfigLongDesc = normalizer.LongDesc(`
+		Generates the kubeconfig file for the admin and for kubeadm itself, and saves it to admin.conf file.
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/kubeconfig.go]: refer to the full path `/etc/kubernetes/admin.conf`?

File: cmd/kubeadm/app/cmd/phases/kubeconfig.go [hunk 5]
```diff
@@ -28,14 +28,58 @@ import (
...
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletKubeconfigLongDesc = normalizer.LongDesc(`
+		Generates the kubeconfig file for the kubelet to use and saves it to kubelet.conf file.
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/kubeconfig.go]: refer to the full path?

File: cmd/kubeadm/app/cmd/phases/kubeconfig.go [hunk 6]
```diff
@@ -28,14 +28,58 @@ import (
...
+		Generates the kubeconfig file for the scheduler to use and saves it to scheduler.conf file.
+		` + cmdutil.AlphaDisclaimer)
+
+	userKubeconfigLongDesc = normalizer.LongDesc(`
+		Outputs a kubeconfig file for an additional user.
+		` + cmdutil.AlphaDisclaimer)
+
+	userKubeconfigExample = normalizer.Examples(`
+		# Outputs a kubeconfig file for an additional user named foo
+		kubeadm alpha phase Kubeconfig user --client-name=foo
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/kubeconfig.go]: nit: small `kubeconfig`

File: cmd/kubeadm/app/cmd/phases/etcd.go [hunk 7]
```diff
@@ -25,14 +25,30 @@ import (
...
+		`)
 )
 
 // NewCmdEtcd return main command for Etcd phase
 func NewCmdEtcd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "etcd",
-		Short: "Generate static pod manifest file for etcd.",
-		RunE:  cmdutil.SubCmdRunE("etcd"),
+		Short: "Generates static Pod manifest file for etcd.",
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/etcd.go]: `Static` or `static` here?

File: cmd/kubeadm/app/cmd/phases/etcd.go [hunk 8]
```diff
@@ -25,14 +25,30 @@ import (
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 	etcdphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/etcd"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	etcdLocalLongDesc = normalizer.LongDesc(`
+		Generates the static Pod manifest file for a local, single-node etcd instance and saves it to etcd.yaml file.
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/etcd.go]: full paths?

File: cmd/kubeadm/app/cmd/phases/addons.go [hunk 9]
```diff
@@ -30,15 +30,39 @@ import (
...
 
 // NewCmdAddon returns the addon Cobra command
 func NewCmdAddon() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "addon <addon-name>",
+		Use:     "addon",
 		Aliases: []string{"addons"},
-		Short:   "Install an addon to a Kubernetes cluster.",
-		RunE:    cmdutil.SubCmdRunE("addon"),
+		Short:   "Installs addons to a Kubernetes cluster",
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/addons.go]: Installs required addons (for passing Conformance tests)?

**Review by luxas — COMMENTED** [review]
some small-ish nits
Only want to know whether we call it `Static Pod` or `static Pod` and if we could point out the working directory for the operation OR write out the full path

On a high level, LGTM though

File: cmd/kubeadm/app/cmd/phases/etcd.go (hunk 7 above)
Comment (fabriziopandini) [reply cmd/kubeadm/app/cmd/phases/etcd.go]: `static Pod`; according to the documentation style guide, `Pod` is an API objects and should be written in camel case like in the API, while `static` is an adjective and follows common grammar rules

**Review by luxas — APPROVED** [review]
/lgtm
/approve no-issue

**Pull request merged**